routing/pathfind.go

```diff
@@ -1300,6 +1304,16 @@ func processNodeForBlindedPath(g Graph, node route.Vertex,
 	// node that can be used for blinded paths
 	err = g.ForEachNodeChannel(node,
 		func(channel *channeldb.DirectedChannel) error {
+			// If this channel was included  into the chained
+			// channel route.
+			chanInSet := restrictions.channelIncomeSet
```

Comment:
It seems to me that this logic will actually produce a path with 2 chained channels if the third one is not found or is not a valid one (not existent), I wonder if we should allow this, I think as soon as the Incoming Channels are specified we should just construct this explicit route.

Reply:
Actually it will produce any path and it will fail. I've just tested it.

Reply:
you mean no route is created if you specify more channels and the last one is not found ?

can you detail your testcase, `minNumberHops` etc ...

Reply:
I mean no route is created. This is the test I did:

'lncli addinvoice --blind --min_real_blinded_hops 3 --num_blinded_hops 3 --blinded_path_income_channel 264982302359552,200111116320768,189116000043009 50000' 

The first two chained channels exist but the third one does not. No route is created.

Reply:
ahh makes sense because of `-min_real_blinded_hops 3` but should we always force the user to also select the right `-min_real_blinded_hops` value ? or should the selection of the incoming channel imply the former, would prefer the latter

Reply:
Yes, `--num_blinded_hops` must be set correctly  ,if it is less than chained channel list size it returns no route. If it is bigger dummy hops are inserted.

`lncli addinvoice --blind --blinded_path_incoming_channel_list 261683767476224,171523813998593 5000` where the second chained channel does not exist, it returns no route (as expected)

`lncli addinvoice --blind --num_blinded_hops 3 --blinded_path_incoming_channel_list 261683767476224,171523813998592,200111116320767 5000`, where the third chained channel does not exist, it returns no route (as expected)

We can make the selection of the incoming chained channel imply the --num_blinded_hops. Do you think an specific error should be triggered if the  `--num_blinded_hops` is specified and it is smaller than the necessary, or just return no routes is ok?

Reply:
what if the third channel does not exist and we select `--num_blinded_hops 2` I think we would get a path with 2 hops not including the last one and also having no error what happened or ?

Reply:
@saubyk what are your thoughts looking at this from the user perspective ?

Reply:
If we select `--num_blinded_hops 2` and give  a third channel, no routes are returned anyway.

Reply:
you mean no route will be returned at all ? But searching through the first 2 would be successful I am curious why not ?

Reply:
> @saubyk what are your thoughts looking at this from the user perspective ?

I am wondering that we are overcomplicating the solution here. The original requirement is to specify the incoming channel on which to receive the payment. Why have we expanded the scope to try and dictate the complete path or specify more hops than just the channel on which the node should receive?

I was thinking that user should just specify which channel to receive the payment on and let the node figure out the rest. Trying to dial in the control too much will add more scope for errors and actually results in poor UX, as the user now needs to find out more information about the network topology around their node.

Besides is there a good explanation of the use case where the user wants to dictate the path in more detail, which the node's pathfinding cannot figure out?
If there is a strong use case to provide more granularity to the user in the path specification, that can also be handled in a subsequent update, rather than trying to cater to every possible use case in the first iteration of this.

My advice is to start simple, add complexity later. Let me know, what you guys think.

Reply:
In regards of your change, you basically put the check to early, thats why routes where not created because the channel count was not increase this has to be done after:

```
	// Keep track of how many incoming channels this node
			// has. We only use a node as an introduction node if it
			// has channels other than the one that lead us to it.
			chanCount++
```
otherwise the first node will never have enough channels as an intro node. 

Given this fact you will get a route if you do this:

```
lncli addinvoice --blind --min_real_blinded_hops 1 --num_blinded_hops 1  --blinded_path_income_channel 694891348819968,694891348819967   500

{
    "r_hash":  "64fb793c8b77e56fb9f805c24b4d16cfad2df0b5bf61553cbbd6f4c414a41910",
    "payment_request":  "lnbcrt5u1pn35fpppp5vnahj0ytwljklw0cqhpykngke7kjmu94has4209m6m6vg99yrygqdqqxqyz5vq5d8qqqqgnqqqqqqzq9tqqqqqqqqqqzycqqqqqppxy70sqqqqqsn4k5k7qzkdgh2qkju4l97a4v7azwqc5k7heqc95mqcpsayywylupqxng35sfferrns4kyr5x94763pznrtw5996kjeg4z2wcp2drr30fvgv5z7qjqlf5csg4wt8n867769xtf0gazk2vl442tlu5yqmsu9ph562v3szux3lf6r2uzjcdadywgcjk9npfvfykjrsmt4skhw5dt7frt566w6q3he0j8em0yaxjemsv2uysfsu4wxyuc2nt4f3vdxdpas6z97pv9h3pe4n72uc32a7k4j8g55metcwfy2m6hy76pd7av9a76t2cwu0darnwpxad9w9u247zw5cxt90hcg2xqmpjgxncknygehg8dxsuznztq5s3vrrg9p4yqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqs9qgq3j59psqxgyn2mwlugf2cukwg42fw7l9m2a6k4mpzuldf22rx3x64f3tceru50pwteqg38xh3k5kxwu89q7qz6gsxwhutlkykq6pfsxspahcgz3",
    "add_index":  "6",
    "payment_addr":  "9a8fe879211540cb422cc39b45776c507680c08a0d35baa45f11e97715d5ecce"
}
```

So I propose just going first with 1 incoming channel, because for the UX we definitely need some improvement to support chained routes

Reply:
So I agree with @saubyk here and fist go with the incoming channel and add complexity later if we need it.

Reply:
hmmm I dont agree - let me chat to ziggie offline a bit